### PR TITLE
Zoom books

### PR DESCRIPTION
**Pull request by davidpmccormick**
Relates to #4442 

Using the `ImageViewer` on the `/item` page to allow e.g. zooming in the same way as `/works`.

- Refactored the `IIIFResponsiveImage` component to take the `src` and `srcSet` props it was previously generating itself from an `imageService` because the current `/works` images don't use an `imageService`. Thinking about it, this means it should probably just be a `ResponsiveImage` component. I think we could/should clean up various uses of `Image` `UiImage` and `IIIFResponsiveImage` to a single pattern.
- The `item` previous/next button controls are currently positioned on top of the zoom control. One of these should move before this is merged.

Screenshot 2019-05-07 at 10 51 02

### CODE REVIEW

File: common/views/components/ImageViewer/ImageViewer.js [hunk 1]
```diff
@@ -147,18 +159,23 @@ const ImageViewer = ({ id, contentUrl, infoUrl, width }: ImageViewerProps) => {
...
+        extraClasses={classNames({
+          'block h-center': true,
+          [spacing({ s: 2 }, { margin: ['bottom'] })]: true,
+          'cursor-zoom-in': viewButtonMounted,
+        })}
+        lang={lang}
+        clickHandler={() => handleViewerDisplay('Image')}
+        alt={
+          (canvasOcr && canvasOcr.replace(/"/g, '')) ||
+          'no text alternative is available for this image'
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]: Out of interest - is this going to be useful, especially if repeated many times?

File: common/views/components/ImageViewer/ImageViewerImage.js [hunk 2]
```diff
@@ -59,13 +64,20 @@ type Props = {|
...
   };
 
   useEffect(() => {
-    setupViewer(infoUrl, id, handleScriptError);
-  }, []);
+    if (viewer) {
+      viewer.destroy();
+    }
+
+    setupViewer(infoUrl, id, handleScriptError).then(viewer => {
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewerImage.js]: You could use await or `setupViewer(infoUrl, id, handleScriptError).then(setViewer)`.
Neither's better, just shorter.

File: common/views/components/ImageViewer/ImageViewerImage.js [hunk 3]
```diff
@@ -59,13 +64,20 @@ type Props = {|
...
+  const [viewer, setViewer] = useState(null);
   const handleScriptError = () => {
     setScriptError(true);
   };
 
   useEffect(() => {
-    setupViewer(infoUrl, id, handleScriptError);
-  }, []);
+    if (viewer) {
+      viewer.destroy();
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewerImage.js]: Do you need to `setViewer(null)`?

File: common/views/components/ImageViewer/ImageViewer.js (hunk 1 above)
Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewer.js]: Good question. We have to output something and we don't have a way to e.g. announce 'no images you're about to look at in this set have alternative text'. But we can shorten it down to `no text alternative` for now. Drafting accessible copy is a good one – I expect we'd need to seek help from the content team.

File: common/views/components/ImageViewer/ImageViewerImage.js (hunk 2 above)
Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewerImage.js]: `useEffect(async () => …` isn't supported and feels like overkill to write an extra async function to wrap `setupViewer`. But I like the shorter `.then(setViewer)`

File: common/views/components/ImageViewer/ImageViewerImage.js (hunk 3 above)
Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewerImage.js]: Good point – thanks.

File: common/views/components/ImageViewer/ImageViewer.js (hunk 1 above)
Comment (jamesgorrie) [reply common/views/components/ImageViewer/ImageViewer.js]: >  from the content team 

Or Orla and her team?

Interesting thought on announcing that there is no text for _any_ image over always announcing it thought, something that could be nice to test.

File: common/views/components/ImageViewer/ImageViewerImage.js (hunk 2 above)
Comment (jamesgorrie) [reply common/views/components/ImageViewer/ImageViewerImage.js]: Ah yes, sorry my bad.

**Comment by davidpmccormick** [comment]
I think the question of how to rationalise our various image components down to less than we currently have can probably wait till after this goes in? Would be good to get to it sooner rather than later though.

**Comment by jamesgorrie** [comment]
> I think the question of how to rationalise our various image components down to less than we currently have can probably wait till after this goes in? Would be good to get to it sooner rather than later though.

Should you create an issue for this, so you can make sure it makes it to planning?

**Comment by davidpmccormick** [comment]
#4449

**Pull request merged**